### DISCUSSION TITLE
Javadoc

### DISCUSSION THREAD

**Discussion by Vliro** [mfussenegger/nvim-jdtls#124] [General]
Does nvim jdtls have javadoc? The autosuggestions in the floating window only show function signature

**Comment by mfussenegger** [comment]
Yes it does: 

![image](https://user-images.githubusercontent.com/38700/130476585-c2c5a2a4-5d2e-4c92-95e1-7d597f19f5a3.png)

signature help is by default disabled. You can enable it if you add the following setting to your config:

```lua
config.settings = {
    java = {
      signatureHelp = { enabled = true },
    }
}
```

**Reply by Vliro** [reply]
Hi, thank you. I will try this. Is this compatible with e.g nvim saga or is its own thing?

**Reply by mfussenegger** [reply]
This is the default `vim.lsp.buf.signature_help()`.  Generally it should work with nvim saga too, but I've no clue why it re-implements the built-in signature help and what kind of changes it made.

**Reply by gnfisher** [reply]
I have signature help enabled as above, but I am still getting "No signature help available" for anything I try it on. Is there something else I might need to tweak to get it working @mfussenegger?

**Reply by gnfisher** [reply]
I got the hover and signature help to show content by installing the `openjdk-11-source` package (oops); however, I see links to  `jdt://...` resources as below. Is there a way to configure jdtls to not include links or to not format with markdown?

```
void java.io.PrintStream.println(String x)
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Prints a String and then terminate the line. This method behaves as though it invokes [print(String)](jdt://contents/java.base/java.io/PrintStream.class?=advent-java/%5C/usr%5C/lib%5C/jvm%5C/java-11-openjdk-amd64%5C/lib%5C/jrt-fs.jar%60java.base=/maven.pomderived=/true=/=/javadoc_location=/https:%5C/%5C/docs.oracle.com%5C/en%5C/java%5C/javase%5C/11%5C/docs%5C/api%5C/=/%3Cjava.io%28PrintStream.class#744) and then [println()](jdt://contents/java.base/java.io/PrintStream.class?=advent-java/%5C/usr%5C/lib%5C/jvm%5C/java-11-openjdk-amd64%5C/lib%5C/jrt-fs.jar%60java.base=/maven.pomderived=/true=/=/javadoc_location=/https:%5C/%5C/docs.oracle.com%5C/en%5C/java%5C/javase%5C/11%5C/docs%5C/api%5C/=/%3Cjava.io%28PrintStream.class#771).

 *  **Parameters:**
    
     *  **x** The `String` to be printed.
```

**Reply by mfussenegger** [reply]
You could add some conceal definitions. For example create a file `lsp_markdown.vim` within `.config/nvim/after/syntax/`:

```vimL
syn clear markdownLink
syn region markdownLink matchgroup=markdownLinkDelimiter start="(" end=")" contains=markdownUrl,mkJdtLink keepend contained
syntax match mkJdtLink /jdt:\/\/.*/ containedin=markdownLink conceal
```

But this may look a bit odd if `wrap` is on, as it will keep line breaks despite no longer being necessary. (See https://github.com/vim/vim/issues/260)

**Reply by egovelox** [reply]
Yes, thanks, it tends to improves readablity a bit.
I think I would like to try improve this, if I can.

@mfussenegger since you wrote in the README

> nvim-jdtls adds some logic to handle jdt:// URIs. 

may I ask you how I can open such an URI, with nvim programmatically ?

**Reply by mfussenegger** [reply]
Via a `BufReadCmd` that calls a function that uses the language server to retrieve the content for the URI. See https://github.com/mfussenegger/nvim-jdtls/blob/e64f05e950e748824bb8b67e89fa4398807a81b4/lua/jdtls.lua#L960-L1022
